noxfile.py

```diff
@@ -215,7 +218,7 @@ def doc(session: nox.Session) -> None:
     if not GCOVR_ISOLATED_TEST and not (
         # Github actions on MacOs can't use Docker
         platform.system() == "Darwin"
-        and "GITHUB_ACTION" in os.environ
+        and CI_RUN is True
```

Comment:
```suggestion
        and CI_RUN
```

Reply:
Even so, the truthy checks are shorter and maybe more pythonic, I learned to hate them over the years because of all the bugs I saw them cause (especially `if obj.method_i_forgot_to_call:` :sweat_smile: ), so I started to use an explicit style. But that is a taste issue, so I will fix it.

Reply:
I want to add pylint in the future. And I thought even flake8 gives an error with the explicit check.

Reply:
I think what you mean is [E712](https://docs.astral.sh/ruff/rules/true-false-comparison/) where one should not use an `==` to compare against singletons like `True`, `False` or `None` but rather use a truthy check or `is` since `is` check for identity and not just equality (`__eq__`).
Personally, I followed the examples of `pydantic` and `FastAPI` to change all my linter+formatter tooling to [ruff](https://docs.astral.sh/ruff/) which replaced `flake8`, `yesqa`, `black`, and `isort` in my python projects. Compared to `pylint` it has the advantages that it runs A LOT faster and it has editor support to autofix some issues.
The other tooling I couldn't live without anymore is [`pre-commit`](https://pre-commit.com/) to manage+run all the code-QA tooling (added bonus is that you won't have things like `run black` or `fix linter issue` in you commit history :sweat_smile: ).